### PR TITLE
SWPWA-1443 - Support for recently viewed products widget

### PR DESCRIPTION
**Pull request by AleksandrsKondratjevs**
At first need to update cms-graphql: 
https://github.com/scandipwa/cms-graphql/pull/14

### CODE REVIEW

File: src/app/route/ProductPage/ProductPage.container.js [hunk 1]
```diff
@@ -262,6 +270,32 @@ export class ProductPageContainer extends PureComponent {
...
+
+        if (recentProducts.length === MAX_NUMBER_OF_RECENT_PRODUCTS) {
+            recentProducts.pop();
+        }
+
+        // Remove product from existing recentProducts to add it later in the beginning
+        const newRecentProducts = recentProducts.filter(({ sku }) => (newSku !== sku));
+
+        newRecentProducts.unshift(product);
+        updateRecentlyViewedProducts(newRecentProducts);
```
Comment (alfredsgenkins) [inline src/app/route/ProductPage/ProductPage.container.js:298]: I think this logic part may be moved to the reducer

File: src/app/route/ProductPage/ProductPage.container.js [hunk 2]
```diff
@@ -262,6 +270,32 @@ export class ProductPageContainer extends PureComponent {
         if (JSON.stringify(items) !== JSON.stringify(prevItems)) {
             this.getRequiredProductOptions(items);
         }
+
+        this._addToRecentlyViewedProducts();
+    }
+
+    _addToRecentlyViewedProducts() {
+        const {
+            product, product: { sku: newSku },
```
Comment (alfredsgenkins) [inline src/app/route/ProductPage/ProductPage.container.js:279]: Move this to new line each if it stays.

File: src/app/component/RecentlyViewedWidget/RecentlyViewedWidget.component.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        return (
+            <ul block="RecentlyViewedWidget" elem="Page">
+                { products.slice(0, pageSize).map((product, i) => (
+                    <ProductCard
+                      selectedFilters={ product.selectedFilters }
+                      product={ product }
+                      // eslint-disable-next-line react/no-array-index-key
+                      key={ `${product.id}_${i}` }
+                    />
+                )) }
```
Comment (alfredsgenkins) [inline src/app/component/RecentlyViewedWidget/RecentlyViewedWidget.component.js:43]: Let's split this into another function

**Review by alfredsgenkins — CHANGES_REQUESTED** [review]
Also please update composer dependency to this module version: https://github.com/scandipwa/cms-graphql/releases/tag/1.4.2

File: src/app/component/RecentlyViewedWidget/RecentlyViewedWidget.config.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+export const RECENTLY_VIEWED_PRODUCTS = 'recentlyViewedProducts';
```
Comment (yeegor) [inline src/app/component/RecentlyViewedWidget/RecentlyViewedWidget.config.js:12]: Prefer:
```js
export const RECENTLY_VIEWED_PRODUCTS = 'RECENTLY_VIEWED_PRODUCTS';
```

**Review by yeegor — APPROVED** [review]
Overall seems legit

**Pull request merged**